### PR TITLE
vscode-extensions.ms-vsliveshare.vsliveshare-pack: Init at 0.4.0

### PR DESCRIPTION
**Pull request by andyrichardson**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by andyrichardson** [comment]
Heads up, this uses the [Microsoft Pre-Release Software License](https://marketplace.visualstudio.com/items/MS-vsliveshare.vsliveshare-pack/license).

File: pkgs/misc/vscode-extensions/default.nix [hunk 1]
```diff
@@ -570,6 +570,15 @@ let
 
       ms-vsliveshare.vsliveshare = callPackage ./ms-vsliveshare-vsliveshare {};
 
+      ms-vsliveshare.vsliveshare-pack = buildVscodeMarketplaceExtension {
+        mktplcRef = {
+          name = "vsliveshare-pack";
+          publisher = "ms-vsliveshare";
+          version = "0.4.0";
+          sha256 = "sha256-xTdfOqdypAaWpmtYIM0H7gwCy1jXNl5+HarvVW/3AiY=";
+        };
```
Comment (SuperSandro2000) [inline pkgs/misc/vscode-extensions/default.nix:579]: Missing meta.

**Comment by andyrichardson** [comment]
Looks like we're building from source for liveshare extensions which is failing on macOS.

Installation using this step seems to work fine without need for build from source - maybe I'm missing something (cc @grahamc)

File: pkgs/misc/vscode-extensions/default.nix (hunk 1 above)
Comment (andyrichardson) [reply pkgs/misc/vscode-extensions/default.nix:579]: [See here](https://github.com/NixOS/nixpkgs/pull/110461#issuecomment-765319578). I would add the license to the project but it isn't listed on spdx 🤔

Comment (SuperSandro2000) [reply pkgs/misc/vscode-extensions/default.nix:579]: You can also add a license object here. Trex coping the definition from the licenses file, change the name and mark it unfree.

**Comment by andyrichardson** [comment]
Closing for now

**Pull request closed without merging**